src/main/java/com/snowplowanalytics/redash/RedashClient.java

```diff
@@ -0,0 +1,437 @@
+        this.client = new OkHttpClient();
+        this.apiKey = apiKey;
+        this.baseUrl = schema + "://" + host + ":" + port + API_PREFIX;
+        this.headers = new Headers.Builder()
+                .add("Accept", "application/json, text/plain, */*")
+                .add("Content-Type", JSON_CONTENT_TYPE)
+                .build();
+    }
+
+    /**
+     * Creates new data-source instance on the Redash server.
+     *
+     * @param dataSource data transfer object which should contain all necessary information.
+     * @return int id of successfully (if so) created instance. In this case object that was transferred as argument
+     * receives that id.
+     * @throws IOException              if server is unable due to connection error or API key is invalid.
+     * @throws IllegalArgumentException if there is already presents data-source on Redash server
+     *                                  with such name
+     */
+    public int createDataSource(DataSource dataSource) throws IOException, IllegalArgumentException {
+        if (isEntityAlreadyExists(getDataSources(), dataSource.getName())) {
+            throw new IllegalArgumentException(DATA_SOURCE_ALREADY_EXISTS);
+        }
+        String url = baseUrl + DATA_SOURCES_URL_PREFIX + API_KEY_URL_PARAM + apiKey;
+        String response = post(url, new Gson().toJson(dataSource), CheckResponseStatus.NO);
+        int id = getIdFromJson(response);
+        dataSource.setId(id);
+        return id;
+    }
+
+    /**
+     * Updates data-source on Redash server with provided as argument DataSource instance. Please note that password field
+     * doesn't take a part in making decision of necessity to update data-source as the server does not return it (password field).
+     *
+     * @param dataSource Argument's name field is used to find specific data-source on the server which (if found) then
+     *                   will be updated by it's fields, if it is necessary.
+     * @return false, if all the fields except password are equals, and there is nothing to update.
+     * True if one or more fields needs to be updated.
+     * @throws IOException with the message that the server response in the case if data-source with provided name
+     *                     couldn't be found, or wrong API key, or there is any connection error.
+     */
+    public boolean updateDataSource(DataSource dataSource) throws IOException {
+        DataSource fromDataBase;
+        try {
+            fromDataBase = getDataSource(dataSource.getName());
+        } catch (IllegalArgumentException e) {
+            throw new IOException(e.getMessage());
+        }
+        if (isDataSourceAlreadyUpToDate(fromDataBase, dataSource)) {
+            return false;
+        }
+        String url = baseUrl + DATA_SOURCES_URL_PREFIX + "/" + fromDataBase.getId() + API_KEY_URL_PARAM + apiKey;
+        post(url, new Gson().toJson(dataSource), CheckResponseStatus.NO);
+        return true;
```

Comment:
How are we assuring that the data-source has actually been updated?  The response does not get checked and we simply default to true?

Reply:
The same as above. If the response status is 200 then we're sure that entity was successfully updated.

Reply:
Awesome so we just need to add this check then!